src/components/cutscenes/PokeCenterCutscene.ts

```diff
@@ -177,6 +177,18 @@ export class PokeCenterCutscene<TGameStartr extends FullScreenPokemon> extends C
                 this.gameStarter.scenePlayer.stopCutscene();
             });
         this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+        const mapLocation = this.gameStarter.itemsHolder.getItem("map");
+        if (mapLocation === "Pallet Town") {
+            this.gameStarter.itemsHolder.setItem("lastPokecenter", {
+            map: this.gameStarter.itemsHolder.getItem("map"),
+            location: "Player's House Door"
+            });
+        } else {
+            this.gameStarter.itemsHolder.setItem("lastPokecenter", {
+            map: this.gameStarter.itemsHolder.getItem("map"),
+            location: "PokeCenter Outside Door"
+            });
+        }
```

Comment:
Another DRY violation (and alignment). Notice how there's only one difference between the two blocks.